ios/RNTrackPlayer/RNTrackPlayer.swift

```diff
@@ -307,19 +307,18 @@ public class RNTrackPlayer: RCTEventEmitter, AudioSessionControllerDelegate {
                                           bookmarkOptions: options["bookmarkOptions"] as? [String: Any])
         }
 
-        if let interval = options["progressUpdateEventInterval"] as? NSNumber, interval.intValue > 0 {
-            shouldEmitUpdateEventInterval = true
-            configureProgressUpdateEvent(interval: interval.doubleValue)
-        } else {
-            shouldEmitUpdateEventInterval = false
-        }
+        configureProgressUpdateEvent(
+            interval: ((options["progressUpdateEventInterval"] as? NSNumber) ?? 0).doubleValue
+        )
 
         resolve(NSNull())
     }
 
     private func configureProgressUpdateEvent(interval: Double) {
-        let time = CMTime(seconds: interval, preferredTimescale: 1)
-        self.player.timeEventFrequency = .custom(time: time)
+        shouldEmitProgressEvent = interval > 0
+        self.player.timeEventFrequency = shouldEmitProgressEvent
+            ? .custom(time: CMTime(seconds: interval, preferredTimescale: 1))
+            : .everySecond
```

Comment:
The big difference I see here is that we're setting the interval to 1s when we cancel, which we weren't doing before. Are we sure this is desired?

Reply:
the reason I am doing this is that the user might have set the interval to something very small temporarily, expecting that setting it back to `0` would return stuff to normal.. (I was going off the comment that says it is impossible to turn it off..)